src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -2363,18 +2363,60 @@ private IEnumerable<PSModuleInfo> CreateFakeModuleObject(IEnumerable<ModuleSpeci
             bool isConsideredCompatible = ModuleUtils.IsPSEditionCompatible(moduleManifestPath, inferredCompatiblePSEditions);
             if (!BaseSkipEditionCheck && !isConsideredCompatible)
             {
-                if (importingModule)
+                if (PowerShellConfig.Instance.IsImplicitWinCompatEnabled())
                 {
-                    IList<PSModuleInfo> moduleProxies = ImportModulesUsingWinCompat(new string [] {moduleManifestPath}, null, new ImportModuleOptions());
+                    if (importingModule)
+                    {
+                        IList<PSModuleInfo> moduleProxies = ImportModulesUsingWinCompat(new string [] {moduleManifestPath}, null, new ImportModuleOptions());
 
-                    // we are loading by a single ManifestPath so expect max of 1
-                    if(moduleProxies.Count > 0)
+                        // we are loading by a single ManifestPath so expect max of 1
+                        if(moduleProxies.Count > 0)
+                        {
+                            return moduleProxies[0];
+                        }
+                        else
+                        {
+                            return null;
+                        }
+                    }
+                }
+                else
+                {
+                    containedErrors = true;
+                    if (writingErrors)
                     {
-                        return moduleProxies[0];
+                        message = StringUtil.Format(
+                            Modules.ImplicitWinCompatDisabled,
+                            moduleManifestPath,
+                            string.Join(',', inferredCompatiblePSEditions));
+
+                        InvalidOperationException ioe = new InvalidOperationException(message);
+
+                        ErrorRecord er = new ErrorRecord(
+                            ioe,
```

Comment:
NIT: just move line 2393 to here rather than creating the `ioe` variable only used for this purpose

Reply:
Updated.